src/domain-engine.js

```diff
@@ -132,10 +132,10 @@ class DomainEngine {
     for (var i = 0, len = handlers.length; i < len; i++) {
       var { key, domain, handler } = handlers[i]
 
-      var last = get(state, key)
+      var last = get(result, key)
       var next = handler.call(domain, last, action.payload)
 
-      result = set(state, key, next)
+      result = set(result, key, next)
```

Comment:
This is the bug. I added `experimental.const_params` to the flow config to prevent assigning over arguments. However I didn't update these internal assignments. The result is that, with multiple domain handlers operating on the same state key, prior operations would be discarded.

💫